flink-streaming-java/src/test/java/org/apache/flink/streaming/api/environment/RemoteStreamExecutionEnvironmentTest.java

```diff
@@ -20,96 +20,84 @@
 public class RemoteStreamExecutionEnvironmentTest extends TestLogger {
 
 	/**
 	 * Verifies that the port passed to the RemoteStreamEnvironment is used for connecting to the cluster.
 	 */
 	@Test
 	public void testPortForwarding() throws Exception {
+		final String host = "fakeHost";
+		final int port = 99;
+		final Configuration conf = new Configuration();
+
+		final RemoteStreamEnvironment env = (RemoteStreamEnvironment) StreamExecutionEnvironment.createRemoteEnvironment(
+			host,
+			port,
+			conf);
+
+		final JobID jobID = new JobID();
+
+		env.setPlanExecutorFactory((_host, _port, _conf) -> {
+			assertThat(_host, is(host));
+			assertThat(_port, is(port));
+			assertThat(_conf, is(conf));
+			return new RemoteExecutor(_host, _port, _conf) {
+				@Override
+				public JobExecutionResult executePlan(Pipeline plan, List<URL> jarFiles, List<URL> classpathList) {
+					return new JobExecutionResult(jobID, 0, Collections.emptyMap());
+				}
+			};
+		});
 
-		String host = "fakeHost";
-		int port = 99;
-		JobID jobID = new JobID();
-		JobResult jobResult = (new JobResult.Builder())
-			.jobId(jobID)
-			.netRuntime(0)
-			.applicationStatus(ApplicationStatus.SUCCEEDED)
-			.build();
-
-		RestClusterClient mockedClient = Mockito.mock(RestClusterClient.class);
-		when(mockedClient.submitJob(any())).thenReturn(CompletableFuture.completedFuture(new JobSubmissionResult(jobID)));
-		when(mockedClient.requestJobResult(any())).thenReturn(CompletableFuture.completedFuture(jobResult));
-
-		PowerMockito.whenNew(RestClusterClient.class).withAnyArguments().thenAnswer((invocation) -> {
-				Object[] args = invocation.getArguments();
-				Configuration config = (Configuration) args[0];
-
-				Assert.assertEquals(host, config.getString(RestOptions.ADDRESS));
-				Assert.assertEquals(port, config.getInteger(RestOptions.PORT));
-				return mockedClient;
-			}
-		);
-
-		final Configuration clientConfiguration = new Configuration();
-		final StreamExecutionEnvironment env = StreamExecutionEnvironment.createRemoteEnvironment(
-			host, port, clientConfiguration);
 		env.fromElements(1).map(x -> x * 2);
-		JobExecutionResult actualResult = env.execute("fakeJobName");
-		Assert.assertEquals(jobID, actualResult.getJobID());
+		JobExecutionResult actualResult = env.execute();
+		assertThat(actualResult.getJobID(), is(jobID));
 	}
 
 	@Test
 	public void testRemoteExecutionWithSavepoint() throws Exception {
-		SavepointRestoreSettings restoreSettings = SavepointRestoreSettings.forPath("fakePath");
-		RemoteStreamEnvironment env = new RemoteStreamEnvironment("fakeHost", 1,
-			null, new String[]{}, null, restoreSettings);
-		env.fromElements(1).map(x -> x * 2);
-
-		JobID jobID = new JobID();
-		JobResult jobResult = (new JobResult.Builder())
-			.jobId(jobID)
-			.netRuntime(0)
-			.applicationStatus(ApplicationStatus.SUCCEEDED)
-			.build();
-
-		RestClusterClient mockedClient = Mockito.mock(RestClusterClient.class);
+		final String host = "fakeHost";
```

Comment:
I just noticed that this test was actually broken by an earlier change. If you look at the original version (https://github.com/apache/flink/commit/bb9b176357bef502391f612de85cce87f45c8d7f) you can see that mockito is used to check whether the right savepoint settings are passed to the client. The current version doesn't do that anymore, this test doesn't really test anything.

Reply:
Yes, I agree this test doesn't test savepoint setting in any way. We possibly just remove it or actually I think it is straight forward a parameter is passed. We don't have to test simple field getter.

Reply:
As @aljoscha already pointed out, the purpose of the test is to assert that the savepoint settings are passed. Removing the test isn't going to achieve that. If it was broken by another change, it needs to be fixed. Maybe there is an easier way to implement it now, but it would be prudent to understand existing code before modifying it.

Reply:
We are able to test savepoint settings passed correctly by retrieving from getter. Whether or not a getter is valuable for testing is another topic.